### PR TITLE
Update config spec to include new fields

### LINKED ISSUE #4129: Updated config spec docs to include context changes
Pach contexts have been extended to [include k8s context data](https://github.com/pachyderm/pachyderm/pull/4058). The [config spec docs](https://docs.pachyderm.io/en/latest/reference/config_spec.html) need to be updated to reflect this.

### PR DESCRIPTION
**Pull request by ysimonson**
Closes #4129 

Blocked by #4442

### CODE REVIEW

File: doc/docs/master/reference/config_spec.md [hunk 1]
```diff
@@ -62,6 +71,11 @@ A map of context names to their configurations. Pachyderm contexts are akin to
 kubernetes contexts (and in fact reference the kubernetes context that they're
 associated with.)
 
+#### Source
+
+An int specifying where the config came from. This is for internal use, and
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
An integer that specifies where the config came from. This parameter is for internal use only and
```

File: doc/docs/master/reference/config_spec.md [hunk 2]
```diff
@@ -62,6 +71,11 @@ A map of context names to their configurations. Pachyderm contexts are akin to
...
+shouldn't be modified.
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
should not be modified.
```

File: doc/docs/master/reference/config_spec.md [hunk 3]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
 
 The currently active transaction for batching together pachctl commands. This
 can be set or cleared via many of the `pachctl * transaction` commands.
+
+#### Cluster name
+
+The name of the underlying kubernetes cluster, extracted from the kubernetes
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
The name of the underlying Kubernetes cluster, extracted from the Kubernetes
```

File: doc/docs/master/reference/config_spec.md [hunk 4]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+context.
+
+#### Auth info
+
+The name of the underlying kubernetes cluster's auth credentials, extracted
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
The name of the underlying Kubernetes cluster's auth credentials, extracted
```

File: doc/docs/master/reference/config_spec.md [hunk 5]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+from the kubernetes context.
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
from the Kubernetes context.
```

File: doc/docs/master/reference/config_spec.md [hunk 6]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+
+#### Namespace
+
+The underlying kubernetes cluster's namespace, extracted from the kubernetes
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
The underlying Kubernetes cluster's namespace, extracted from the Kubernetes
```

File: doc/docs/master/reference/config_spec.md [hunk 7]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+context.
+
+#### Cluster ID
+
+The pachyderm cluster ID - used to ensure we're running operations on the
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
The pachyderm cluster ID that is used to ensure the operations run on the
```

File: doc/docs/master/reference/config_spec.md [hunk 8]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+expected cluster.
+
+#### Port forwarders
+
+A mapping of service name -> local port. This field is populated when running
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
A mapping of `service name -> local port`. This field is populated when you run
```

File: doc/docs/master/reference/config_spec.md [hunk 9]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+explicit port forwarding (`pachctl port-forward`), so that subsequent
+`pachctl` operations know to use the explicit port forwarder.
+
+This field will be removed when the `pachctl port-forward` operation
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
This field is removed when the `pachctl port-forward` operation
```

File: doc/docs/master/reference/config_spec.md [hunk 10]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+completes, but note that you may need to manually delete the field from your
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
completes. You might need to manually delete the field from your
```

File: doc/docs/master/reference/config_spec.md [hunk 11]
```diff
@@ -85,3 +99,33 @@ if pachctl actions are authorized. This is only set when auth is enabled.
...
+config if the process was killed being being able to do so.
```
Comment (svekars) [inline doc/docs/master/reference/config_spec.md]:
```suggestion
config if the process failed to remove the field automatically.
```

**Comment by ysimonson** [comment]
@svekars this is unblocked and ready for another pass

**Pull request merged**